lib/private/Security/CSP/ContentSecurityPolicyNonceManager.php

```diff
@@ -58,7 +58,11 @@ public function __construct(CsrfTokenManager $csrfTokenManager,
 	 */
 	public function getNonce(): string {
 		if($this->nonce === '') {
-			$this->nonce = base64_encode($this->csrfTokenManager->getToken()->getEncryptedValue());
+			if (empty($_SERVER['CSP_NONCE'])) {
```

Comment:
```suggestion
			if (empty($this->request->server['CSP_NONCE'])) {
```

Is a bit more elegant IMO

Reply:
If that's the typical approach for the project, then it's fine by me.

Reply:
OK, that now fails the tests, so this variable is not directly accessing the $_SERVER var. How would I test this correctly then?

Reply:
```
Index: tests/lib/Security/CSP/ContentSecurityPolicyNonceManagerTest.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tests/lib/Security/CSP/ContentSecurityPolicyNonceManagerTest.php	(date 1563390270000)
+++ tests/lib/Security/CSP/ContentSecurityPolicyNonceManagerTest.php	(date 1563396922000)
@@ -21,10 +21,10 @@
 
 namespace Test\Security\CSP;
 
+use OC\AppFramework\Http\Request;
 use OC\Security\CSP\ContentSecurityPolicyNonceManager;
 use OC\Security\CSRF\CsrfToken;
 use OC\Security\CSRF\CsrfTokenManager;
-use OCP\IRequest;
 use Test\TestCase;
 
 class ContentSecurityPolicyNonceManagerTest extends TestCase  {
@@ -32,13 +32,13 @@
 	private $csrfTokenManager;
 	/** @var ContentSecurityPolicyNonceManager */
 	private $nonceManager;
+	/** @var Request */
+	private $request;
 
 	public function setUp() {
 		$this->csrfTokenManager = $this->createMock(CsrfTokenManager::class);
-		$this->nonceManager = new ContentSecurityPolicyNonceManager(
-			$this->csrfTokenManager,
-			$this->createMock(IRequest::class)
-		);
+		$this->request = $this->createMock(Request::class);
+		$this->nonceManager = new ContentSecurityPolicyNonceManager($this->csrfTokenManager, $this->request);
 	}
 
 	public function testGetNonce() {
@@ -59,11 +59,18 @@
 
 	public function testGetNonceServerVar() {
 		$token = 'SERVERNONCE';
-		$_SERVER['CSP_NONCE'] = $token;
+
+		$this->request
+			->method('__isset')
+			->with('server')
+			->willReturn(true);
+
+		$this->request
+			->method('__get')
+			->with('server')
+			->willReturn(['CSP_NONCE' => $token]);
 
 		$this->assertSame($token, $this->nonceManager->getNonce());
 		$this->assertSame($token, $this->nonceManager->getNonce());
-		
-		unset($_SERVER['CSP_NONCE']);
 	}
 }

```